proxy/src/telemetry/metrics/latency.rs

```diff
@@ -1,229 +1,76 @@
-    Latency(500),
+    Bucket::Le(100),
+    Bucket::Le(200),
+    Bucket::Le(300),
+    Bucket::Le(400),
+    Bucket::Le(500),
     // prometheus.LinearBuckets(100, 100, 5),
-    Latency(1_000),
-    Latency(2_000),
-    Latency(3_000),
-    Latency(4_000),
-    Latency(5_000),
+    Bucket::Le(1_000),
+    Bucket::Le(2_000),
+    Bucket::Le(3_000),
+    Bucket::Le(4_000),
+    Bucket::Le(5_000),
     // prometheus.LinearBuckets(1000, 1000, 5),
-    Latency(10_000),
-    Latency(20_000),
-    Latency(30_000),
-    Latency(40_000),
-    Latency(50_000),
+    Bucket::Le(10_000),
+    Bucket::Le(20_000),
+    Bucket::Le(30_000),
+    Bucket::Le(40_000),
+    Bucket::Le(50_000),
     // prometheus.LinearBuckets(10000, 10000, 5),
-    Latency(100_000),
-    Latency(200_000),
-    Latency(300_000),
-    Latency(400_000),
-    Latency(500_000),
-    // Prometheus implicitly creates a max bucket for everything that
-    // falls outside of the highest-valued bucket, but we need to
-    // create it explicitly.
-    Latency(u32::MAX),
-];
-
-/// A series of latency values and counts.
+    Bucket::Le(100_000),
+    Bucket::Le(200_000),
+    Bucket::Le(300_000),
+    Bucket::Le(400_000),
+    Bucket::Le(500_000),
+    // A final upper bound.
+    Bucket::Inf,
+]);
+
+/// A duration in milliseconds.
 #[derive(Debug, Default, Clone)]
-pub struct Histogram {
-
-    /// Array of buckets in which to count latencies.
-    ///
-    /// The upper bound of a given bucket `i` is given in `BUCKET_BOUNDS[i]`.
-    buckets: [Counter; NUM_BUCKETS],
-
-    /// The total sum of all observed latency values.
-    ///
-    /// Histogram sums always explicitly wrap on overflows rather than
-    /// panicking in debug builds. Prometheus' [`rate()`] and [`irate()`]
-    /// queries handle breaks in monotonicity gracefully (see also
-    /// [`resets()`]), so wrapping is less problematic than panicking in this
-    /// case.
-    ///
-    /// Note, however, that Prometheus actually represents this using 64-bit
-    /// floating-point numbers. The correct semantics are to ensure the sum
-    /// always gets reset to zero after Prometheus reads it, before it would
-    /// ever overflow a 52-bit `f64` mantissa.
-    ///
-    /// [`rate()`]: https://prometheus.io/docs/prometheus/latest/querying/functions/#rate()
-    /// [`irate()`]: https://prometheus.io/docs/prometheus/latest/querying/functions/#irate()
-    /// [`resets()`]: https://prometheus.io/docs/prometheus/latest/querying/functions/#resets
-    ///
-    // TODO: Implement Prometheus reset semantics correctly, taking into
-    //       consideration that Prometheus represents this as `f64` and so
-    //       there are only 52 significant bits.
-    pub sum: Wrapping<u64>,
-}
-
-/// A latency in tenths of a millisecond.
-#[derive(Debug, Default, Eq, PartialEq, Ord, PartialOrd, Copy, Clone, Hash)]
-pub struct Latency(u32);
-
-
-// ===== impl Histogram =====
-
-impl Histogram {
+pub struct Ms(pub Duration);
```

Comment:
This is great and I love it 👍 thanks for finally doing away with the old and crufty "decimilliseconds" code